src/wasm-interpreter.h

```diff
@@ -1426,24 +1435,36 @@ class ExpressionRunner : public OverriddenVisitor<SubType, Flow> {
       cast.outcome = cast.Null;
       return cast;
     }
-    // The input may not be a struct or an array; for example it could be an
+    // The input may not be GC data or a function; for example it could be an
     // anyref of null (already handled above) or anything else (handled here,
     // but this is for future use as atm the binaryen interpreter cannot
     // represent external references).
-    if (!cast.originalRef.isData()) {
+    if (!cast.originalRef.isData() && !cast.originalRef.isFunction()) {
       cast.outcome = cast.Failure;
       return cast;
     }
-    auto gcData = cast.originalRef.getGCData();
-    auto refRtt = gcData->rtt;
-    auto intendedRtt = rtt.getSingleValue();
-    if (!refRtt.isSubRtt(intendedRtt)) {
-      cast.outcome = cast.Failure;
+    Literal seenRtt;
+    Literal intendedRtt = rtt.getSingleValue();
+    if (cast.originalRef.isFunction()) {
+      // Function casts are simple in that they have no RTT hierarchies; instead
+      // each reference has the canonical RTT for the signature.
+      auto* func = module->getFunction(cast.originalRef.getFunc());
```

Comment:
Does this need to handle the case where `module` is null?

Reply:
I don't think we can handle that case. I can add an assertion here, would be better than a crash I suppose, though I believe all relevant code paths are known to be safe statically.

Reply:
should we do something like `return (NONCONSTANT_FLOw)` so the program can continue?

Reply:
In case of not having a module? I don't think we should try to handle that case, it should be a fatal error. We'd need to fix the code path leading to here so it provides a module.